scripts/bootstrap.sh.in

```diff
@@ -144,56 +144,77 @@ retry_dns_count: 3
+            --label io.kubernetes.container.name=salt-master \
+            --state Running)"
+    done
+    echo "$salt_container"
 }
 
 sync_salt() {
-    $SALT_MASTER_CALL salt '*' saltutil.sync_all refresh=True \
+    local salt_master_call
+    salt_master_call="crictl exec -i $(get_salt_container)"
+
+    $salt_master_call salt '*' saltutil.sync_all refresh=True \
+        saltenv=metalk8s-@@VERSION
+
+    $salt_master_call salt '*' mine.update
+
+    $salt_master_call salt-run saltutil.sync_all saltenv=metalk8s-@@VERSION
+    $salt_master_call salt-run saltutil.sync_roster saltenv=metalk8s-@@VERSION
+    $salt_master_call salt-run metalk8s_saltutil.sync_auth \
         saltenv=metalk8s-@@VERSION
-    $SALT_MASTER_CALL salt '*' mine.update
-    local SALT_CONTAINER
-    SALT_CONTAINER="$(crictl ps -q \
-            --label io.kubernetes.pod.namespace=kube-system \
-            --label io.kubernetes.container.name=salt-master \
-            --state Running)"
-    test -n "$SALT_CONTAINER"
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run saltutil.sync_all saltenv=metalk8s-@@VERSION
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run saltutil.sync_roster saltenv=metalk8s-@@VERSION
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run metalk8s_saltutil.sync_auth saltenv=metalk8s-@@VERSION
 }
 
 orchestrate_bootstrap() {
-    # TODO: FIX register IP
-    local cidr
-    local registry_ip
-    cidr=$(salt-call --local pillar.get networks:control_plane --out=txt | cut -f 2 -d ' ')
-    registry_ip=$(salt-call --local network.ip_addrs cidr="$cidr" --out=txt | sed "s/.*'\(\([0-9]\{1,3\}\.\)\{3\}[0-9]\{1,3\}\)'.*/\1/")
     # Grains must be set (in `/etc/salt/grains`) *before* invoking `salt-call`,
     # otherwise grains set during execution won't be taken into account
     # properly.
     ${SALT_CALL} --local --retcode-passthrough state.sls \
         metalk8s.node.grains \
         saltenv=metalk8s-@@VERSION \
         pillarenv=metalk8s-@@VERSION
+
+    
+    # TODO: Registry IP should not need a pillar override here
+    local cidr
+    local registry_ip
+    cidr=$( \
+        ${SALT_CALL} --local pillar.get networks:control_plane --out=txt \
+        | cut -f 2 -d ' ' \
+    )
+    registry_ip=$( \
+        ${SALT_CALL} --local network.ip_addrs cidr="$cidr" --out=txt \
+        | sed "s/.*'\(\([0-9]\{1,3\}\.\)\{3\}[0-9]\{1,3\}\)'.*/\1/" \
+    )
     ${SALT_CALL} --local --retcode-passthrough state.orchestrate \
         metalk8s.orchestrate.bootstrap_without_master \
         saltenv=metalk8s-@@VERSION \
         pillarenv=metalk8s-@@VERSION \
-        pillar="{'registry_ip': $registry_ip}"
-    set_salt_command
+        pillar="{'registry_ip': '$registry_ip'}"
+
     sync_salt
-    # shellcheck disable=SC2086
-    ${SALT_MASTER_CALL} salt-run state.orchestrate \
+
+    local salt_master_call
+    salt_master_call="crictl exec -i $(get_salt_container)"
+
+    # Wait for all minions to respond before proceeding
```

Comment:
Why `all`? Don't we only need bootstrap? So `salt-call test.ping` should do, no?

Reply:
So:

- Yes we do only need bootstrap, I could do `salt-run metalk8s_saltutil.wait_minions tgt="$(cat /etc/salt/minion_id)"`, but I found it to be unnecessary (I'll let you choose)
- A simple `salt '*' test.ping` will always have return code 0, even if minions fail to respond, so we would have needed to parse the output of the command (I found it to be easier in Python)
- I didn't try to `salt-call test.ping` on the minion directly, maybe it waits better than `salt '*' test.ping`, and has an appropriate return code
- With a runner like this, we could include it in an orchestrate (e.g. the one used to add new nodes to the cluster), while a simple `test.ping` may be harder to use (with retries and expected response...)

Reply:
> Yes we do only need bootstrap, I could do salt-run metalk8s_saltutil.wait_minions tgt="$(cat /etc/salt/minion_id)", but I found it to be unnecessary (I'll let you choose)

I guess that's a bit of an issue in an environment where
- we need to restore the `bootstrap` node for whatever reason (i.e. re-run this script)
- for some reason, not all minions are/can be online

Reply:
OK, I'll add the more specific `tgt` then.